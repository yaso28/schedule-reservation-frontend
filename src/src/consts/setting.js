export default Object.freeze({
  reservation: Object.freeze({
    category: 'reservation',
    mail_to: 'mail_to',
    mail_subject: 'mail_subject',
    mail_message_begin: 'mail_message_begin',
    mail_message_end: 'mail_message_end'
  }),
  reservation_public: Object.freeze({
    category: 'reservation_public',
    notes: 'notes'
  })
})
