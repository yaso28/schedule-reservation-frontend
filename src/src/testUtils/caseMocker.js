import _ from 'lodash'

class CaseMocker {
  mock (caseList) {
    return jest.fn((...args) => {
      for (let i = 0; i < caseList.length; i++) {
        const currentCase = caseList[i]
        if (_.isEqual(args, currentCase.args)) {
          return currentCase.value
        }
      }
      throw new Error('caseMocker.mock: no args matched')
    })
  }
}

export default new CaseMocker()
