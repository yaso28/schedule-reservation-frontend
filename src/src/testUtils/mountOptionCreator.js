import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
