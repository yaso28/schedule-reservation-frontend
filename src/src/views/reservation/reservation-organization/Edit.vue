<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyText
            :label="$t('reservation_organization.registration_number')"
            :form="form"
            field="registration_number"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ReservationOrganizationEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const record = await this.apiService.getReservationOrganization(this.id)
        this.form = this.apiService.makeForm({
          name: record.name,
          abbreviation: record.abbreviation,
          registration_number: record.registration_number
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateReservationOrganization(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.reservation-organization.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
