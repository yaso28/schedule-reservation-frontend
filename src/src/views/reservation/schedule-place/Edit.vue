<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyNumber
            :label="$t('schedule_place.price_per_hour')"
            :form="form"
            field="price_per_hour"
            :min="0"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'SchedulePlaceEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MyNumber: require('@/components/inputs/MyNumber').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const record = await this.apiService.getSchedulePlace(this.id)
        this.form = this.apiService.makeForm({
          name: record.name,
          abbreviation: record.abbreviation,
          price_per_hour: record.price_per_hour
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateSchedulePlace(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-place.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
