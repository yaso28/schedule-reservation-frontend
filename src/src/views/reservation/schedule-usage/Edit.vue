<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="loaded"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MySwitch
            :label="$t('schedule_usage.is_public')"
            :form="form"
            field="is_public"
          />
          <MySelect
            :label="$t('reservation_organization.model')"
            :form="form"
            field="reservation_organization_id"
            :options="reservationOrganizations"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ScheduleUsageEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MySwitch: require('@/components/inputs/MySwitch').default,
    MySelect: require('@/components/inputs/MySelect').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null,
      reservationOrganizations: null
    }
  },
  computed: {
    loaded () {
      return !!(this.form && this.reservationOrganizations)
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        let record
        [this.reservationOrganizations, record] = await Promise.all([
          this.reservationOrganizations ?? this.apiService.getReservationOrganizationList(),
          this.apiService.getScheduleUsage(this.id)
        ])
        this.form = this.apiService.makeForm({
          name: record.name,
          is_public: record.is_public,
          reservation_organization_id: record.reservation_organization_id
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateScheduleUsage(this.id, this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-usage.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
