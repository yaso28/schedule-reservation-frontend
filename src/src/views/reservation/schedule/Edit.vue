<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="loaded"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyDate
            :label="$t('schedule.ymd')"
            :form="form"
            field="ymd"
            :show-clear="false"
          />
          <MyTime
            :label="$t('schedule.begins_at')"
            :form="form"
            field="begins_at"
          />
          <MyTime
            :label="$t('schedule.ends_at')"
            :form="form"
            field="ends_at"
          />
          <MySelect
            :label="$t('schedule_place.model')"
            :form="form"
            field="schedule_place_id"
            :options="placeList"
          />
          <MySelect
            :label="$t('schedule_usage.model')"
            :form="form"
            field="schedule_usage_id"
            :options="usageList"
          />
          <MySelect
            :label="$t('schedule_timetable.model')"
            :form="form"
            field="schedule_timetable_id"
            :options="timetableList"
          />
          <MySelect
            :label="$t('schedule_status.model')"
            :form="form"
            field="schedule_status_id"
            :options="scheduleStatusList"
          />

          <hr class="my-4">

          <p class="h5">
            {{ $t('reservation.model') }}
          </p>
          <CCard
            v-for="(reservation, index) in form.values.reservation_list"
            :key="reservation.id"
            class="my-2"
          >
            <CCardBody>
              <MyTime
                :label="$t('schedule.begins_at')"
                :form="form"
                :field="`reservation_list.${index}.begins_at`"
              />
              <MyTime
                :label="$t('schedule.ends_at')"
                :form="form"
                :field="`reservation_list.${index}.ends_at`"
              />
              <MySelect
                :label="$t('reservation_status.model')"
                :form="form"
                :field="`reservation_list.${index}.reservation_status_id`"
                :options="reservationStatusList"
              />
              <CLink
                class="btn btn-outline-primary"
                :to="{ name: 'reservation.reservation.split', params: { id: reservation.id } }"
              >
                {{ $t('action.split') }}
              </CLink>
            </CCardBody>
          </CCard>
        </CCardBody>
        <CCardFooter>
          <CButton
            class="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ReservationScheduleEdit',
  components: {
    MyDate: require('@/components/inputs/MyDate').default,
    MyTime: require('@/components/inputs/MyTime').default,
    MySelect: require('@/components/inputs/MySelect').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  props: {
    id: {
      type: [Number, String],
      required: true
    }
  },
  data () {
    return {
      form: null,
      placeList: null,
      usageList: null,
      timetableList: null,
      scheduleStatusList: null,
      reservationStatusList: null
    }
  },
  computed: {
    loaded () {
      return !!(this.form && this.placeList && this.usageList && this.timetableList && this.scheduleStatusList && this.reservationStatusList)
    }
  },
  watch: {
    id () {
      this.load()
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        let scheduleRecord
        let reservationRecordList
        [this.placeList, this.usageList, this.timetableList, this.scheduleStatusList, this.reservationStatusList, scheduleRecord, reservationRecordList] = await Promise.all([
          this.placeList ?? this.apiService.getSchedulePlaceList(),
          this.usageList ?? this.apiService.getScheduleUsageList(),
          this.timetableList ?? this.apiService.getScheduleTimetableList(),
          this.scheduleStatusList ?? this.apiService.getScheduleStatusList(),
          this.reservationStatusList ?? this.apiService.getReservationStatusList(),
          this.apiService.getSchedule(this.id),
          this.apiService.getReservationListForSchedule(this.id)
        ])
        this.form = this.apiService.makeForm({
          ymd: scheduleRecord.ymd,
          begins_at: scheduleRecord.begins_at,
          ends_at: scheduleRecord.ends_at,
          schedule_place_id: scheduleRecord.schedule_place_id,
          schedule_usage_id: scheduleRecord.schedule_usage_id,
          schedule_timetable_id: scheduleRecord.schedule_timetable_id,
          schedule_status_id: scheduleRecord.schedule_status_id,
          reservation_list: reservationRecordList.map(record => ({
            id: record.id,
            begins_at: record.begins_at,
            ends_at: record.ends_at,
            reservation_status_id: record.reservation_status_id
          }))
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.updateSchedule(this.id, this.form)
        this.$store.dispatch('alert/saveSuccessNoRedirect')
        this.form.setInit()
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
