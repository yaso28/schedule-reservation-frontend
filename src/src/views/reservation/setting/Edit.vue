<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="loaded"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <div
            v-for="setting in settingList"
            :key="`${setting.category}-${setting.key}`"
            v-c-tooltip="{
              content: setting.record.description,
              placement: 'top'
            }"
          >
            <MyText
              v-if="setting.rows <= 1"
              :label="$t(setting.label)"
              :form="setting.form"
              field="value"
              :id-prefix="`${setting.category}-${setting.key}`"
            />
            <MyTextarea
              v-if="setting.rows > 1"
              :label="$t(setting.label)"
              :form="setting.form"
              field="value"
              :id-prefix="`${setting.category}-${setting.key}`"
              :rows="setting.rows"
            />
          </div>
        </CCardBody>
        <CCardFooter>
          <CButton
            class="btn-reset"
            color="secondary"
            @click="reset"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
import setting from '@/consts/setting'

export default {
  name: 'ReservationSettingEdit',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MyTextarea: require('@/components/inputs/MyTextarea').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      settingList: [
        [setting.reservation.category, setting.reservation.mail_to, 'send.mail_to', 1],
        [setting.reservation.category, setting.reservation.mail_subject, 'send.subject', 1],
        [setting.reservation.category, setting.reservation.mail_message_begin, 'send.message_begin', 5],
        [setting.reservation_public.category, setting.reservation_public.notes, 'common.notes', 5],
        [setting.reservation.category, setting.reservation.mail_message_end, 'send.message_end', 5]
      ].map(([category, key, label, rows]) => ({ category, key, label, rows, record: null, form: null })),
      loaded: false
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const records = await Promise.all(this.settingList.map(setting => this.apiService.getSetting(setting.category, setting.key)))
        for (let i = 0; i < this.settingList.length; i++) {
          this.settingList[i].record = records[i]
          this.settingList[i].form = this.apiService.makeForm({ value: records[i].value })
        }
        this.loaded = true
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    reset () {
      for (let i = 0; i < this.settingList.length; i++) {
        this.settingList[i].form.reset()
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await Promise.all(this.settingList.map(setting => this.apiService.updateSetting(setting.category, setting.key, setting.form)))
        this.$store.dispatch('alert/saveSuccessNoRedirect')
        for (let i = 0; i < this.settingList.length; i++) {
          this.settingList[i].form.setInit()
        }
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
