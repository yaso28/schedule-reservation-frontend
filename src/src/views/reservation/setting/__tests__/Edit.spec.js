
import Edit from '../Edit'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import promiseMocker from '@/testUtils/promiseMocker'
import storeModules from '@/store/modules'
import myFormMocker from '@/testUtils/myFormMocker'
import caseMocker from '@/testUtils/caseMocker'
import setting from '@/consts/setting'

jest.mock('@/store/modules')
jest.mock('@/models/MyForm')

const makeSettingRecord = (categoryName, keyName, value, description) => ({ category_name: categoryName, key_name: keyName, value, description })
const makeEditTarget = (name, expectedRecord) => ({ name, expectedRecord, makeFormArgs: { value: expectedRecord.value } })
const editTargetList = [
  makeEditTarget(
    '宛先メールアドレス',
    makeSettingRecord(setting.reservation.category, setting.reservation.mail_to, 'sample@example.com', '練習予定を送信する宛先のメールアドレスです。')
  ),
  makeEditTarget(
    '件名',
    makeSettingRecord(setting.reservation.category, setting.reservation.mail_subject, '{month_name}の練習予定', 'メールの件名です。{month_name}は"yyyy年mm月"に置き換わります。')
  ),
  makeEditTarget(
    '本文（開始）',
    makeSettingRecord(setting.reservation.category, setting.reservation.mail_message_begin, 'こんにちは。\n{month_name}の練習予定をご連絡します。', 'メール本文の最初に載せる文章です。{month_name}は"yyyy年mm月"に置き換わります。この文章のあとに予定が載ります。')
  ),
  makeEditTarget(
    '本文（終了）',
    makeSettingRecord(setting.reservation.category, setting.reservation.mail_message_end, '以上です。\nよろしくお願いします。', 'メール本文の最後に載せる文章です。')
  ),
  makeEditTarget(
    '注意事項',
    makeSettingRecord(setting.reservation_public.category, setting.reservation_public.notes, '〇〇にご注意ください。\n××にご注意ください。', 'メール本文の途中（予定のあと）に載せる注意事項の文章です。また、公開している練習予定ページの注意事項にも同じ文章が載ります。')
  )
]
const editTargetEach = editTargetList.map(target => [target.name, target])
const addToEditTargetList = addProc => {
  for (let i = 0; i < editTargetList.length; i++) {
    addProc(editTargetList[i])
  }
}

describe('Edit', () => {
  let mountOption
  let apiService
  beforeEach(() => {
    promiseMocker.init()
    mountOption = mountOptionCreator.create(storeModules)
    apiService = mountOption.provide.apiService
  })

  afterEach(() => {
    promiseMocker.clear()
    jest.resetAllMocks()
  })

  describe('マウント', () => {
    let wrapper
    beforeEach(async () => {
      addToEditTargetList(target => {
        target.loadCase = {
          args: [target.expectedRecord.category_name, target.expectedRecord.key_name],
          value: target.expectedRecord,
          error: new Error(`load:${target.name}`),
          switch: promiseMocker.createSwitch()
        }
      })
      apiService.getSetting = promiseMocker.mockForCases(editTargetList.map(target => target.loadCase))
      wrapper = await pageTester.mount(Edit, { name: 'reservation.setting.edit' }, mountOption)
    })

    test('titleタグ', () => {
      pageTester.assertTitleTag()
    })

    test('metaタグ', () => {
      pageTester.assertMetaTags()
    })

    test('Processingカバー表示開始', () => {
      expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
    })

    test.each(editTargetEach)('レコード取得処理呼出：%s', (name, target) => {
      expect(apiService.getSetting).toHaveBeenCalledWith(target.expectedRecord.category_name, target.expectedRecord.key_name)
    })

    test('レコード取得処理呼出：回数', () => {
      expect(apiService.getSetting).toHaveBeenCalledTimes(editTargetList.length)
    })

    test('Processingカバー表示中', () => {
      expect(storeModules.processing.mutations.endGet).not.toHaveBeenCalled()
    })

    describe.each(editTargetEach)('レコード取得：エラー：%s', (name, target) => {
      beforeEach(() => {
        promiseMocker.setSwitch(target.loadCase.switch, false)
        apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
        promiseMocker.flushPendings()
      })

      test('エラー処理', () => {
        expect(apiService.isHandledError).toHaveBeenCalledWith(target.loadCase.error)
      })

      test('Processingカバー表示終了', () => {
        expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
      })
    })

    describe('レコード取得：成功', () => {
      beforeEach(() => {
        addToEditTargetList(target => {
          target.form = myFormMocker.createMock(target.makeFormArgs)
        })
        apiService.makeForm = caseMocker.mock(editTargetList.map(target => ({ args: [target.makeFormArgs], value: target.form })))
        promiseMocker.flushPendings()
      })

      test.each(editTargetEach)('フォーム作成：%s', (name, target) => {
        expect(apiService.makeForm).toHaveBeenCalledWith(target.makeFormArgs)
      })

      test('フォーム作成：回数', () => {
        expect(apiService.makeForm).toHaveBeenCalledTimes(editTargetList.length)
      })

      test('ページ全体表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })

      test('Processingカバー表示終了', () => {
        expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
      })

      describe('リセットボタンクリック', () => {
        beforeEach(() => {
          wrapper.find('button.btn-reset').trigger('click')
        })

        test.each(editTargetEach)('フォームリセット：%s', (name, target) => {
          expect(target.form.reset).toHaveBeenCalledWith()
        })
      })

      describe('フォーム送信', () => {
        beforeEach(() => {
          addToEditTargetList(target => {
            target.saveCase = {
              args: [target.expectedRecord.category_name, target.expectedRecord.key_name, target.form],
              value: null,
              error: new Error(`save:${target.name}`),
              switch: promiseMocker.createSwitch()
            }
          })
          apiService.updateSetting = promiseMocker.mockForCases(editTargetList.map(target => target.saveCase))
          wrapper.find('form').trigger('submit')
        })

        test('Processingカバー表示開始', () => {
          expect(storeModules.processing.mutations.beginPost).toHaveBeenCalled()
        })

        test('アラートクリア', () => {
          expect(storeModules.alert.mutations.clear).toHaveBeenCalled()
        })

        test.each(editTargetEach)('保存処理呼出：%s', (name, target) => {
          expect(apiService.updateSetting).toHaveBeenCalledWith(target.expectedRecord.category_name, target.expectedRecord.key_name, target.form)
        })

        test('保存処理呼出：回数', () => {
          expect(apiService.updateSetting).toHaveBeenCalledTimes(editTargetList.length)
        })

        test('Processingカバー表示中', () => {
          expect(storeModules.processing.mutations.endPost).not.toHaveBeenCalled()
        })

        describe.each(editTargetEach)('保存：エラー：%s', (name, target) => {
          beforeEach(() => {
            promiseMocker.setSwitch(target.saveCase.switch, false)
            apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
            promiseMocker.flushPendings()
          })

          test('エラー処理', () => {
            expect(apiService.isHandledError).toHaveBeenCalledWith(target.saveCase.error)
          })

          test('Processingカバー表示終了', () => {
            expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
          })

          test('成功アラートは表示しない', () => {
            expect(storeModules.alert.actions.saveSuccessNoRedirect).not.toHaveBeenCalled()
          })

          test.each(editTargetEach)('フォーム初期値更新は呼び出さない：%s', (name, target) => {
            expect(target.form.setInit).not.toHaveBeenCalled()
          })
        })

        describe('保存：成功', () => {
          beforeEach(() => {
            promiseMocker.flushPendings()
          })

          test('成功アラート表示', () => {
            expect(storeModules.alert.actions.saveSuccessNoRedirect).toHaveBeenCalled()
          })

          test.each(editTargetEach)('フォーム初期値更新：%s', (name, target) => {
            expect(target.form.setInit).toHaveBeenCalledWith()
          })

          test('Processingカバー表示終了', () => {
            expect(storeModules.processing.mutations.endPost).toHaveBeenCalled()
          })
        })
      })
    })
  })
})
