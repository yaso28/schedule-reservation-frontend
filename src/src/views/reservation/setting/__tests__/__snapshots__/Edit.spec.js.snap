// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit マウント metaタグ 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="2"
    name="robots"
  />,
]
`;

exports[`Edit マウント titleタグ 1`] = `"やそスポーツクラブ | 設定"`;

exports[`Edit マウント レコード取得：成功 ページ全体表示 1`] = `
<div>
  <h1>
    設定
  </h1>
   
  <form
    class=""
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div>
          <div
            class="form-group form-row"
            role="group"
          >
            <label
              class="col-form-label col-sm-3"
              for="reservation-mail_to-value"
            >
               宛先メールアドレス 
            </label>
            <div
              class="col-sm-9"
            >
              <!---->
              <input
                class="form-control"
                id="reservation-mail_to-value"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
        <div>
          <div
            class="form-group form-row"
            role="group"
          >
            <label
              class="col-form-label col-sm-3"
              for="reservation-mail_subject-value"
            >
               件名 
            </label>
            <div
              class="col-sm-9"
            >
              <!---->
              <input
                class="form-control"
                id="reservation-mail_subject-value"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <!---->
        </div>
        <div>
          <!---->
           
          <div
            class="form-group form-row"
            role="group"
          >
            <label
              class="col-form-label col-sm-3"
              for="reservation-mail_message_begin-value"
            >
               本文（開始） 
            </label>
            <div
              class="col-sm-9"
            >
              <!---->
              <textarea
                class="form-control"
                id="reservation-mail_message_begin-value"
                rows="5"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div>
          <!---->
           
          <div
            class="form-group form-row"
            role="group"
          >
            <label
              class="col-form-label col-sm-3"
              for="reservation_public-notes-value"
            >
               注意事項 
            </label>
            <div
              class="col-sm-9"
            >
              <!---->
              <textarea
                class="form-control"
                id="reservation_public-notes-value"
                rows="5"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div>
          <!---->
           
          <div
            class="form-group form-row"
            role="group"
          >
            <label
              class="col-form-label col-sm-3"
              for="reservation-mail_message_end-value"
            >
               本文（終了） 
            </label>
            <div
              class="col-sm-9"
            >
              <!---->
              <textarea
                class="form-control"
                id="reservation-mail_message_end-value"
                rows="5"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </div>
       
      <footer
        class="card-footer"
      >
        <button
          class="btn btn-reset btn-secondary"
          type="button"
        >
          
          リセット
        
        </button>
         
        <button
          class="btn btn-primary"
          type="submit"
        >
          
          保存
        
        </button>
      </footer>
    </div>
  </form>
</div>
`;
